components/servo/examples/winit_minimal.rs

```diff
@@ -123,46 +125,69 @@ impl ApplicationHandler<WakerEvent> for App {
         if let Self::Running {
             window_delegate,
             servo,
+            webviews,
         } = self
         {
-            let mut events_for_servo = vec![];
-            for (_webview_id, message) in servo.get_events() {
+            for (_webview_id, message) in servo.get_events().collect::<Vec<_>>() {
                 match message {
                     // FIXME: rust-analyzer autocompletes this as top_level_browsing_context_id
                     EmbedderMsg::WebViewOpened(webview_id) => {
+                        // TODO: We currently assume `webview` refers to the same webview as `_webview_id`
                         let rect = window_delegate.get_coordinates().get_viewport().to_f32();
-                        events_for_servo.extend([
-                            EmbedderEvent::FocusWebView(webview_id),
-                            EmbedderEvent::MoveResizeWebView(webview_id, rect),
-                            EmbedderEvent::RaiseWebViewToTop(webview_id, true),
-                        ]);
+                        if let Some(webview) =
+                            webviews.iter().find(|webview| webview.id() == webview_id)
+                        {
+                            webview.focus();
+                            webview.move_resize(rect);
+                            webview.raise_to_top(true);
+                        }
+                    },
+                    EmbedderMsg::AllowOpeningWebView(webview_id_sender) => {
+                        let webview = servo.new_auxiliary_webview();
+                        let _ = webview_id_sender.send(Some(webview.id()));
+                        webviews.push(webview);
+                    },
+                    EmbedderMsg::AllowNavigationRequest(pipeline_id, _) => {
+                        servo.handle_events([EmbedderEvent::AllowNavigationResponse(
+                            pipeline_id,
+                            true,
+                        )]);
                     },
                     _ => {},
                 }
             }
-            servo.handle_events(events_for_servo);
+            // FIXME: still needed for the compositor to actually run
+            servo.handle_events([]);
         }
         match event {
             WindowEvent::CloseRequested => {
-                if matches!(self, Self::Running { .. }) {
-                    let Self::Running { servo, .. } = replace(self, Self::Exiting) else {
-                        unreachable!()
-                    };
-                    // TODO: ask Servo to shut down and wait for EmbedderMsg::Shutdown?
-                    servo.deinit();
```

Comment:
I hope eventually we can use drop trait to handle all these, including webrender's deinit.